config.mk

```diff
@@ -0,0 +1,217 @@
+
+else
+#############################################################################
+# Help message
+#############################################################################
+ifeq (help,$(COMPILER))
+
+$(info COMPILER can be one of: gnu, arm, fujitsu, cray)
+$(info  * gnu: GNU Compiler)
+$(info  * arm: Arm Compiler for Linux)
+$(info  * fujitsu: Fujitsu Compiler)
+$(info  * cray: Cray Compiler)
+$(error )
+
+else
+#############################################################################
+# Error handling
+#############################################################################
+
+$(error Invalid parameter: COMPILER=$(COMPILER))
+
+#############################################################################
+endif
+endif
+endif
+endif
+endif
+#############################################################################
+
+
+#############################################################################
+# Generic CBLAS library flags
+#############################################################################
+ifeq (none,$(LIBRARY))
+
+CFLAGS_LIBRARY   = 
+CXXFLAGS_LIBRARY = 
+FFLAGS_LIBRARY   = 
+LDFLAGS_LIBRARY  = -lm
+
+else
+#############################################################################
+# Arm Performance Library (ArmPL) library flags
+#############################################################################
+ifeq (armpl,$(LIBRARY))
+
+ifeq (arm,$(COMPILER))
+# The Arm compiler automatically knows where to find ArmPL and will 
+# automatically enable OpenMP if the -fopenmp flag is present
+CFLAGS_LIBRARY   = -armpl -DUSE_ARMPL
+CXXFLAGS_LIBRARY = -armpl -DUSE_ARMPL
+FFLAGS_LIBRARY   = -armpl -DUSE_ARMPL
+LDFLAGS_LIBRARY  = -armpl
+else
+# If not using the Arm compiler, make sure to load the ArmPL module.
+# ARMPL_INCLUDES and ARMPL_LIBRARIES are set by the ArmPL modulefile
+CFLAGS_LIBRARY   = -I$(ARMPL_INCLUDES) -DUSE_ARMPL
+CXXFLAGS_LIBRARY = -I$(ARMPL_INCLUDES) -DUSE_ARMPL
+FFLAGS_LIBRARY   = -I$(ARMPL_INCLUDES) -DUSE_ARMPL
+LDFLAGS_LIBRARY  = -L$(ARMPL_LIBRARIES) -larmpl -lamath -lastring
+endif
+
+else
+#############################################################################
+# Fujitsu SSL II library flags
+#############################################################################
+ifeq (ssl2,$(LIBRARY))
+
+ifeq (fujitsu,$(COMPILER))
+CFLAGS_LIBRARY   = -SSL2 -DUSE_SSL2
+CXXFLAGS_LIBRARY = -SSL2 -DUSE_SSL2
+FFLAGS_LIBRARY   = -SSL2 -DUSE_SSL2
+LDFLAGS_LIBRARY  = -SSL2
+else
+$(error SSL2 can only be used with the Fujitsu compiler)
+endif
+
+else
+#############################################################################
+# Error handling
+#############################################################################
+
+$(error Invalid parameter: LIBRARY=$(LIBRARY))
+
+#############################################################################
+endif
+endif
+endif
+#############################################################################
+
+ifeq (neon,$(VEC_OPT))
```

Comment:
Where is this VEC_FLAG getting used? Looks like it isn't referenced. Please can you clean-up? Thanks. Or add code for manual SVE/ Neon vectorization.